libraries/app/database_api.cpp

```diff
@@ -921,71 +921,98 @@ std::map<std::string, full_account> database_api_impl::get_full_accounts( const
       std::for_each(asset_range.first, asset_range.second,
-                    [&acnt] (const asset_object& asset) {
-                       acnt.assets.emplace_back(asset.id);
-                    });
+            [&acnt, api_limit_get_full_accounts] (const asset_object& asset) {
+               if(acnt.assets.size() < api_limit_get_full_accounts)
+                  acnt.assets.emplace_back(asset.id);
+            });
 
       // get withdraws permissions
-      auto withdraw_range = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().equal_range(account->id);
-      std::for_each(withdraw_range.first, withdraw_range.second,
-                    [&acnt] (const withdraw_permission_object& withdraw) {
-                       acnt.withdraws.emplace_back(withdraw);
-                    });
+      auto withdraw_indices = _db.get_index_type<withdraw_permission_index>().indices();
+      auto withdraw_from_range = withdraw_indices.get<by_from>().equal_range(account->id);
+      std::for_each(withdraw_from_range.first, withdraw_from_range.second,
+            [&acnt, api_limit_get_full_accounts] (const withdraw_permission_object& withdraw) {
+               if(acnt.withdraws.size() < api_limit_get_full_accounts)
+                  acnt.withdraws.emplace_back(withdraw);
+            });
+      auto withdraw_authorized_range = withdraw_indices.get<by_authorized>().equal_range(account->id);
+      std::for_each(withdraw_authorized_range.first, withdraw_authorized_range.second,
+            [&acnt, api_limit_get_full_accounts] (const withdraw_permission_object& withdraw) {
+               if((std::find(acnt.withdraws.begin(), acnt.withdraws.end(), withdraw) == acnt.withdraws.end()) or
+                     (acnt.withdraws.size() < api_limit_get_full_accounts))
+                  acnt.withdraws.emplace_back(withdraw);
+            });
 
       // get htlcs
       auto htlc_from_range = _db.get_index_type<htlc_index>().indices().get<by_from_id>().equal_range(account->id);
       std::for_each(htlc_from_range.first, htlc_from_range.second,
-            [&acnt] (const htlc_object& htlc) {
-               acnt.htlcs.emplace_back(htlc);
+            [&acnt, api_limit_get_full_accounts] (const htlc_object& htlc) {
+               if(acnt.htlcs.size() < api_limit_get_full_accounts)
+                  acnt.htlcs.emplace_back(htlc);
             });
       auto htlc_to_range = _db.get_index_type<htlc_index>().indices().get<by_to_id>().equal_range(account->id);
       std::for_each(htlc_to_range.first, htlc_to_range.second,
-            [&acnt] (const htlc_object& htlc) {
-               if ( std::find(acnt.htlcs.begin(), acnt.htlcs.end(), htlc) == acnt.htlcs.end() )
+            [&acnt, api_limit_get_full_accounts] (const htlc_object& htlc) {
+               if ((std::find(acnt.htlcs.begin(), acnt.htlcs.end(), htlc) == acnt.htlcs.end()) or
```

Comment:
Using `find` inside the `for_each` loop leads to quadratic complexity. Since this is only useful for the abnormal case of an htlc with `from == to` I suggest to drop the `find` entirely.